### ISSUE TITLE
Allow to disable hierarchy

### ISSUE THREAD

**Issue by griesi007** [boxblinkracer/phpunuhi#55]
I just came across this promising project. I found one issue why it does not work in my case. I was just wondering if there is a solution I just did not found. My json files are plain key value pairs. The keys  can contain sentences. When I run the command to fix the structure, all my translations are split up into multiple levels. So far as I can see this is due to the StorageHierarchy used in the storage and which  seems to not be changeable. Is there any solution to keep my translation flat not split into multiple levels? I am not sure if this is intentional behaviour or a bug. Can this be disabled? So far as I understood I need to run fix structure so that empty translations are created for missing entries which then can be translated afterwards

Below is a sample
Before fixing structure: 
`"For the payroll accounting we need the completely filled out and signed personnel questionnaire from you. All fields marked with a red star or bordered in green are required for printing. After saving, please download the personnel questionnaire from the overview and then upload it, signed, to your profile.": "Per la contabilità salariale abbiamo bisogno del vostro questionario del personale debitamente compilato e firmato. Tutti i campi contrassegnati da una stella rossa o bordati di verde sono obbligatori per la stampa. Dopo il salvataggio, scarica il questionario del personale dalla panoramica e poi caricalo, firmato, sul tuo profilo.`

After fixing structure
```
  "For the payroll accounting we need the completely filled out and signed personnel questionnaire from you": {
    " All fields marked with a red star or bordered in green are required for printing": {
      " After saving, please download the personnel questionnaire from the overview and then upload it, signed, to your profile": {
        "": "Per la contabilità salariale abbiamo bisogno del vostro questionario del personale debitamente compilato e firmato. Tutti i campi contrassegnati da una stella rossa o bordati di verde sono obbligatori per la stampa. Dopo il salvataggio, scarica il questionario del personale dalla panoramica e poi caricalo, firmato, sul tuo profilo."
      }
    }
  },
```

**Comment by boxblinkracer** [comment]
Hi there
thank you for this
Ill get back to you after vacation :)

**Comment by boxblinkracer** [comment]
im so so sorry for the delay

so what i dont understand in your case

usually you have key value pairs in translations

in your case its just a list of translations? can you help me a bit why it is like this?

**Comment by griesi007** [comment]
Have you read my comment? I have a json file with plain key -> value pairs. When running  phpunuhi "fix structure" command, a hierarchy is introduced which is wrong. The json file should stay plain key value pairs. Just try it with the attached json file
[it.json](https://github.com/user-attachments/files/17610180/it.json)

**Comment by boxblinkracer** [comment]
yeah but i think the sample is misleading to me

are your keys really long sentences?
then this is the "problem" :)

**Comment by griesi007** [comment]
@boxblinkracer yes the key can be a sentence. Its the fallback if no translation is defined. A fix should be easy when "StorageHierarchy" could be made configurable and thus be turned off by configuration. Any chance to get this implemented?

**Comment by boxblinkracer** [comment]
ah ok

yes supporting should be possible

but can i also ask you if that is some default behavior of a language or is it a custom thing for you?

because usually technical keys are used like labelTitle, etc. 
if the key is already a sentence of an existing translation value that seems to be a bit risky. if the original sentence is changed then the fallback key is not found anymore

just trying to understand this :)

**Comment by griesi007** [comment]
@boxblinkracer I do not think this is language or framework specific.  It´s just a matter how you handle translations. I prefer to not use abstract keys like labelTitle as this makes the code where it is used harder to read and also displays meaningless "code" to users when no translation for the key exists. It´s easier if keys in real language (usually english) are used which of course could be a sentence then. If no translation exists this is what is shown to the user and always makes sense. The worst thing that can happen here is that an english sentence is shown to a user expecting another language. So in my opinion more fault tolerant.

**Comment by boxblinkracer** [comment]
Hi thanks for this
so if you have a full sentence in base language english and all zour languages use this sebtence as key. then you always need to make sure to change the sentence in all files when you adjust the text?

anyway. ill try to support it :)

**Comment by griesi007** [comment]
@boxblinkracer yes that´s the trade off. But I don´t need a english translation :-)

**Comment by boxblinkracer** [comment]
ok now i got it

but yeah. i think i can support it. try to add it to the upcoming release. changelog is already huge and really excited to release it

**Comment by boxblinkracer** [comment]
alright
so the problem was not really the hierarchy, but that the delimiter "." existed in the keys. this made it actually to build nesting levels.

at the moment there is no case to adjust the delimiter which could fix your issue
but instead ive created a new setting that is more to what you thought the issue is, and what you want to archieve

<json nested="false"

this new attribute avoids hierarchy creation

can you try this with your file and let me know if it works?
thank you

**Comment by griesi007** [comment]
@boxblinkracer thanks a lot. It works as expected

**Comment by boxblinkracer** [comment]
released in v1.22.0
